src/core/scheduler_utils.cc

```diff
@@ -47,6 +47,10 @@ InitRequiredEqualInputs(
       required_equal_inputs->emplace(
           std::piecewise_construct, std::forward_as_tuple(input->Name()),
           std::forward_as_tuple(input, itr->second));
+    } else {
+      required_equal_inputs->emplace(
```

Comment:
For perf reasons it would be good not to do this unless the model had optional inputs.